distributed/core.py

```diff
@@ -422,7 +430,7 @@ async def handle_comm(self, comm):
 
         logger.debug("Connection from %r to %s", address, type(self).__name__)
         self._comms[comm] = op
-        await self
```

Comment:
This `await self` without the guard I introduced in `__await__` would allow an incoming comm to restart a currently shutting down scheduler. There is a time window while the scheduler is handling its affairs, before the actual server closes, where it still accepts incoming comms. If during this time a new comm is opened, the scheduler is restarted although it is currently trying to shutdown.

https://github.com/dask/distributed/blob/d9bc3c634a4a3983cdc75e5c39cbdac2afbf6184/distributed/scheduler.py#L3759-L3808

Reply:
I didn't add a dedicated test for this since there are actually many failing tests once the `close` coroutines are shielded